### ISSUE TITLE
Error in setting  networks status with delegate plugin that returns empty Result

### ISSUE THREAD

**Issue by ahalim-intel** [intel/multus-cni#211]
**What happend**:
The sriov cni plugin with DPDK mode has no IPAM config. Therefore returns empty Result. Multus throwing in error as it tries to build network status from this empty Result.

**What you expected to happen**:
Multus should skip delegate results with no IPConfig when aggregating delegate results.

**How to reproduce it (as minimally and precisely as possible)**:
Attache any network CRDs without any IPAM config.

**Anything else we need to know?**:

**Environment**:

- Multus version 
Latest version.
Works fine until 707007abba91b305cb8f946058610891d19b98b5

  image path and image ID (from 'docker images')
- Kubernetes version (use `kubectl version`):
v1.14.0-alpha.0
- Primary CNI for Kubernetes cluster:

- OS (e.g. from /etc/os-release):
CentOS 7
Kernel 4.13.7-1.el7.elrepo.x86_64
- File of '/etc/cni/net.d/'
```
# cat /etc/cni/net.d/00-multus.conf

{
   "name":"defaultnetwork",
   "type":"multus",
   "LogLevel": "debug",
   "LogFile": "/var/log/multus.log",
   "kubeconfig": "/etc/kubernetes/node-kubeconfig.yaml",
   "delegates": [{"cniVersion": "0.3.0", "name":"defaultnetwork", "type": "flannel", "isDefaultGateway": true}],
   "confDir": "/etc/cni/multus/net.d"
}
```
- File of '/etc/cni/multus/net.d'
- NetworkAttachment info (use `kubectl get net-attach-def -o yaml`)
```
# kubectl get net-attach-def -o yaml
apiVersion: v1
items:
- apiVersion: k8s.cni.cncf.io/v1
  kind: NetworkAttachmentDefinition
  metadata:
    creationTimestamp: "2018-12-11T20:31:16Z"
    generation: 1
    name: sriov-dpdk
    namespace: default
    resourceVersion: "602527"
    selfLink: /apis/k8s.cni.cncf.io/v1/namespaces/default/network-attachment-definitions/sriov-dpdk
    uid: b5a50e62-fd83-11e8-a46e-408d5c537d23
  spec:
    config: '{ "type": "sriov", "name": "sriov-dpdk-net", "master": "enp2s0f0", "dpdk":
      { "kernel_driver": "i40evf", "dpdk_driver": "igb_uio", "dpdk_tool":"/root/dpdk/usertools/dpdk-devbind.py"
      } }'
- apiVersion: k8s.cni.cncf.io/v1
  kind: NetworkAttachmentDefinition
  metadata:
    annotations:
      k8s.v1.cni.cncf.io/resourceName: intel.com/sriov_net_A
    creationTimestamp: "2018-12-03T15:33:22Z"
    generation: 1
    name: sriov-net-a
    namespace: default
    resourceVersion: "4735"
    selfLink: /apis/k8s.cni.cncf.io/v1/namespaces/default/network-attachment-definitions/sriov-net-a
    uid: c4c1c803-f710-11e8-a0de-f63fe80d5632
  spec:
    config: '{ "type": "sriov", "name": "sriov-network", "ipam": { "type": "host-local",
      "subnet": "10.56.217.0/24", "routes": [{ "dst": "0.0.0.0/0" }], "gateway": "10.56.217.1"
      } }'
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""

```
- Target pod yaml info (with annotation, use `kubectl get pod <podname> -o yaml`)

```
# kubectl get pod testdpdkpod -o yaml                                                                                                                      
apiVersion: v1                                                                                                                                                                             
kind: Pod                                                                                                                                                                                  
metadata:                                                                                                                                                                                  
  annotations:                                                                                                                                                                             
    k8s.v1.cni.cncf.io/networks: sriov-dpdk                                                                                                                                                
    k8s.v1.cni.cncf.io/networks-status: ""                                                                                                                                                 
  creationTimestamp: "2018-12-12T09:57:57Z"                                                                                                                                                
  labels:                                                                                                                                                                                  
    env: test                                                                                                                                                                              
  name: testdpdkpod                                                                                                                                                                        
  namespace: default                                                                                                                                                                       
  resourceVersion: "665704"                                                                                                                                                                
  selfLink: /api/v1/namespaces/default/pods/testdpdkpod                                                                                                                                    
  uid: 6708f766-fdf4-11e8-a46e-408d5c537d23                                                                                                                                                
spec:                                                                                                                                                                                      
  containers:                                                                                                                                                                              
  - args:                                                                                                                                                                                  
    - while true; do sleep 300000; done;                                                                                                                                                   
    command:                                                                                                                                                                               
    - /bin/bash                                                                                                                                                                            
    - -c                                                                                                                                                                                   
    - --                                                                                                                                                                                   
    image: centos/tools                                                                                                                                                                    
    imagePullPolicy: IfNotPresent                                                                                                                                                          
    name: appcntr1                                                                                                                                                                         
    resources: {}                                                                                                                                                                          
    terminationMessagePath: /dev/termination-log                                                                                                                                           
    terminationMessagePolicy: File                                                                                                                                                         
    volumeMounts:                                                                                                                                                                          
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount                                                                                                                             
      name: default-token-cnp7g                                                                                                                                                            
      readOnly: true                                                                                                                                                                       
  dnsPolicy: ClusterFirst                                                                                                                                                                  
  enableServiceLinks: true                                                                                                                                                                 
  nodeName: silpixa00396659.ir.intel.com                                                                                                                                                   
  nodeSelector:                                                                                                                                                                            
    kubernetes.io/hostname: silpixa00396659.ir.intel.com                                                                                                                                   
  restartPolicy: Always                                                                                                                                                                    
  schedulerName: default-scheduler                                                                                                                                                         
  securityContext: {}
  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 30
  volumes:
  - name: default-token-cnp7g
    secret:
      defaultMode: 420
      secretName: default-token-cnp7g
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2018-12-12T09:57:57Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2018-12-12T09:57:57Z"
    message: 'containers with unready status: [appcntr1]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2018-12-12T09:57:57Z"
    message: 'containers with unready status: [appcntr1]'
    reason: ContainersNotReady
    status: "False"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2018-12-12T09:57:57Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - image: centos/tools
    imageID: ""
    lastState: {}
    name: appcntr1
    ready: false
    restartCount: 0
    state:
      waiting:
        reason: ContainerCreating
  hostIP: 10.237.222.170
  phase: Pending
  qosClass: BestEffort
  startTime: "2018-12-12T09:57:57Z"
```
- Other log outputs (if you use multus logging)

```
# kubectl describe pod testdpdkpod
.
.
.
Node-Selectors:  kubernetes.io/hostname=silpixa00396659.ir.intel.com                                                                                                                       
Tolerations:     <none>                                                                                                                                                                    
Events:                                                                                                                                                                                    
  Type     Reason                  Age                    From                                   Message                                                                                   
  ----     ------                  ----                   ----                                   -------                                                                                   
  Normal   Scheduled               33m                    default-scheduler                      Successfully assigned default/testdpdkpod to silpixa00396659.ir.intel.com                 
  Warning  FailedCreatePodSandBox  32m                    kubelet, silpixa00396659.ir.intel.com  Failed create pod sandbox: rpc error: code = Unknown desc = [failed to set up sandbox container "bf613a84877df18208949685c1f3345c8a10bfffce0065f2e46f1631d7a6c39f" network for pod "testdpdkpod": NetworkPlugin cni failed to set up pod "testdpdkpod_default" network: Multus: Err in setting  networks status: error convert the type.Result to current.Result: cannot convert: no valid IP addresses, failed to clean up sandbox container "bf613a84877df18208949685c1f3345c8a10bfffce0065f2e46f1631d7a6c39f" network for pod "testdpdkpod": NetworkPlugin cni failed to teardown pod "testdpdkpod_default" network: Multus: error in invoke Delegate del - "sriov": DPDK: failed to reset vlan tag for vf 0: resource temporarily unavailable] 
```

Mutus log:
```
2018-12-12T10:05:19Z [debug] validateIfName: /proc/16145/ns/net, eth0
 4272 2018-12-12T10:05:19Z [debug] LoadNetworkStatus: Interfaces:[{Name:cni0 Mac:ce:dc:03:07:32:20 Sandbox:} {Name:veth0b1ef4a2 Mac:7e:2d:6c:23:18:68 Sandbox:} {Name:eth0 Mac:ea:f1:eb:4b:      7e:89 Sandbox:/proc/16145/ns/net}], IP:[{Version:4 Interface:0xc420336388 Address:{IP:192.168.76.36 Mask:ffffff00} Gateway:192.168.76.1}], Routes:[{Dst:{IP:192.168.0.0 Mask:              ffff0000} GW:<nil>}], DNS:{Nameservers:[] Domain: Search:[] Options:[]}, defaultnetwork, true
 4273 2018-12-12T10:05:19Z [debug] getIfname: &{{ sriov-dpdk-net sriov map[] {} {[]  [] []}} {  []}    false false [123 32 34 116 121 112 101 34 58 32 34 115 114 105 111 118 34 44 32 34        110 97 109 101 34 58 32 34 115 114 105 111 118 45 100 112 100 107 45 110 101 116 34 44 32 34 109 97 115 116 101 114 34 58 32 34 101 110 112 50 115 48 102 48 34 44 32 34 100 112 100       107 34 58 32 123 32 34 107 101 114 110 101 108 95 100 114 105 118 101 114 34 58 32 34 105 52 48 101 118 102 34 44 32 34 100 112 100 107 95 100 114 105 118 101 114 34 58 32 34 105         103 98 95 117 105 111 34 44 32 34 100 112 100 107 95 116 111 111 108 34 58 34 47 114 111 111 116 47 100 112 100 107 47 117 115 101 114 116 111 111 108 115 47 100 112 100 107 45 100       101 118 98 105 110 100 46 112 121 34 32 125 32 125]}, eth0, 1
 4274 2018-12-12T10:05:19Z [debug] LoadCNIRuntimeConf: &{8c5295cad0a7bd82add45e890cff66ca49c7fe66319b9866e4a2642a971d3229 /proc/16145/ns/net eth0 IgnoreUnknown=1;                               K8S_POD_NAMESPACE=default;K8S_POD_NAME=testdpdkpod;K8S_POD_INFRA_CONTAINER_ID=8c5295cad0a7bd82add45e890cff66ca49c7fe66319b9866e4a2642a971d3229 /opt/cni/bin [123 34 76 111 103 70          105 108 101 34 58 34 47 118 97 114 47 108 111 103 47 109 117 108 116 117 115 46 108 111 103 34 44 34 76 111 103 76 101 118 101 108 34 58 34 100 101 98 117 103 34 44 34 99 110 105         86 101 114 115 105 111 110 34 58 34 34 44 34 99 111 110 102 68 105 114 34 58 34 47 101 116 99 47 99 110 105 47 109 117 108 116 117 115 47 110 101 116 46 100 34 44 34 100 101 108          101 103 97 116 101 115 34 58 91 123 34 99 110 105 86 101 114 115 105 111 110 34 58 34 48 46 51 46 48 34 44 34 105 115 68 101 102 97 117 108 116 71 97 116 101 119 97 121 34 58 116         114 117 101 44 34 110 97 109 101 34 58 34 100 101 102 97 117 108 116 110 101 116 119 111 114 107 34 44 34 116 121 112 101 34 58 34 102 108 97 110 110 101 108 34 125 93 44 34 107          117 98 101 99 111 110 102 105 103 34 58 34 47 101 116 99 47 107 117 98 101 114 110 101 116 101 115 47 110 111 100 101 45 107 117 98 101 99 111 110 102 105 103 46 121 97 109 108 34        44 34 110 97 109 101 34 58 34 100 101 102 97 117 108 116 110 101 116 119 111 114 107 34 44 34 116 121 112 101 34 58 34 109 117 108 116 117 115 34 125]}, &{{true} <nil> testdpdkpod        default 8c5295cad0a7bd82add45e890cff66ca49c7fe66319b9866e4a2642a971d3229}, net1, <nil>
 4275 2018-12-12T10:05:19Z [debug] delegateAdd: <nil>, net1, &{{ sriov-dpdk-net sriov map[] {} {[]  [] []}} {  []}    false false [123 32 34 116 121 112 101 34 58 32 34 115 114 105 111         118 34 44 32 34 110 97 109 101 34 58 32 34 115 114 105 111 118 45 100 112 100 107 45 110 101 116 34 44 32 34 109 97 115 116 101 114 34 58 32 34 101 110 112 50 115 48 102 48 34 44         32 34 100 112 100 107 34 58 32 123 32 34 107 101 114 110 101 108 95 100 114 105 118 101 114 34 58 32 34 105 52 48 101 118 102 34 44 32 34 100 112 100 107 95 100 114 105 118 101 114       34 58 32 34 105 103 98 95 117 105 111 34 44 32 34 100 112 100 107 95 116 111 111 108 34 58 34 47 114 111 111 116 47 100 112 100 107 47 117 115 101 114 116 111 111 108 115 47 100          112 100 107 45 100 101 118 98 105 110 100 46 112 121 34 32 125 32 125]}, &{8c5295cad0a7bd82add45e890cff66ca49c7fe66319b9866e4a2642a971d3229 /proc/16145/ns/net net1 [[IgnoreUnknown        1] [K8S_POD_NAMESPACE default] [K8S_POD_NAME testdpdkpod] [K8S_POD_INFRA_CONTAINER_ID 8c5295cad0a7bd82add45e890cff66ca49c7fe66319b9866e4a2642a971d3229]] map[] }, /opt/cni/bin
 4276 2018-12-12T10:05:19Z [debug] validateIfName: /proc/16145/ns/net, net1
 4277 2018-12-12T10:05:19Z [debug] LoadNetworkStatus: DNS:{Nameservers:[] Domain: Search:[] Options:[]}, sriov-dpdk-net, false
 4278 2018-12-12T10:05:19Z [error] error convert the type.Result to current.Result: cannot convert: no valid IP addresses
 4279 2018-12-12T10:05:19Z [error] Multus: Err in setting  networks status: error convert the type.Result to current.Result: cannot convert: no valid IP addresses
 4280 2018-12-12T10:05:19Z [debug] LoadDelegateNetConf: {"cniVersion":"0.3.0","isDefaultGateway":true,"name":"defaultnetwork","type":"flannel"}, <nil>,•
 4281 2018-12-12T10:05:19Z [debug] GetK8sArgs: &{8c5295cad0a7bd82add45e890cff66ca49c7fe66319b9866e4a2642a971d3229 /proc/16145/ns/net eth0 IgnoreUnknown=1;K8S_POD_NAMESPACE=default;             K8S_POD_NAME=testdpdkpod;K8S_POD_INFRA_CONTAINER_ID=8c5295cad0a7bd82add45e890cff66ca49c7fe66319b9866e4a2642a971d3229 /opt/cni/bin [123 34 76 111 103 70 105 108 101 34 58 34 47 118        97 114 47 108 111 103 47 109 117 108 116 117 115 46 108 111 103 34 44 34 76 111 103 76 101 118 101 108 34 58 34 100 101 98 117 103 34 44 34 99 110 105 86 101 114 115 105 111 110 34       58 34 34 44 34 99 111 110 102 68 105 114 34 58 34 47 101 116 99 47 99 110 105 47 109 117 108 116 117 115 47 110 101 116 46 100 34 44 34 100 101 108 101 103 97 116 101 115 34 58 91        123 34 99 110 105 86 101 114 115 105 111 110 34 58 34 48 46 51 46 48 34 44 34 105 115 68 101 102 97 117 108 116 71 97 116 101 119 97 121 34 58 116 114 117 101 44 34 110 97 109 101        34 58 34 100 101 102 97 117 108 116 110 101 116 119 111 114 107 34 44 34 116 121 112 101 34 58 34 102 108 97 110 110 101 108 34 125 93 44 34 107 117 98 101 99 111 110 102 105 103         34 58 34 47 101 116 99 47 107 117 98 101 114 110 101 116 101 115 47 110 111 100 101 45 107 117 98 101 99 111 110 102 105 103 46 121 97 109 108 34 44 34 110 97 109 101 34 58 34 100        101 102 97 117 108 116 110 101 116 119 111 114 107 34 44 34 116 121 112 101 34 58 34 109 117 108 116 117 115 34 125]}

```

**Comment by cliffburdick** [comment]
@ahalim-intel sorry, commented in the wrong spot instead of on the issue itself:

do you know when this will be pushed upstream? I ran into this issue with the SRIOV plugin.

**Comment by dougbtv** [comment]
@cliffburdick -- have you tried the `nfvpe/multus:snapshot` image -- in theory #212 was merged earlier this week and should be in that image.

**Comment by cliffburdick** [comment]
@dougbtv trying right now...

**Comment by cliffburdick** [comment]
@dougbtv I'm getting this error:

  Warning  FailedCreatePodSandBox  104s                kubelet, s011  Failed create pod sandbox: rpc error: code = Unknown desc = [failed to set up sandbox container "b6f68bb9599da2f46dc898c8a9f1ba3d35279e294b1915bdece6e05b7578f545" network for pod "ut-cburdick-b6f192": NetworkPlugin cni failed to set up pod "ut-cburdick-b6f192_default" network: netplugin failed but error parsing its diagnostic message "": unexpected end of JSON input, failed to clean up sandbox container "b6f68bb9599da2f46dc898c8a9f1ba3d35279e294b1915bdece6e05b7578f545" network for pod "ut-cburdick-b6f192": NetworkPlugin cni failed to teardown pod "ut-cburdick-b6f192_default" network: netplugin failed but error parsing its diagnostic message "": unexpected end of JSON input]

my network attachment looks like this, and I think it's valid:
apiVersion: "k8s.cni.cncf.io/v1"
kind: NetworkAttachmentDefinition
metadata:
  name: host-device-conf1
spec:
  config: '{
            "cniVersion": "0.2.0",
            "name": "mynet1",
            "type": "sriov",
            "master": "ens1f0",
            "l2enable": true
        }'

**Comment by ahalim-intel** [comment]
@cliffburdick As @dougbtv pointed out, this issue should be resolved now with PR #212 . From your last logs it seems some issue with sriov-cni config. We need more contexts on this. Are you using this without [sriov device plugin](https://github.com/intel/sriov-network-device-plugin)?

**Comment by s1061123** [comment]
This is already addressed, so closed.
